### PR TITLE
[3.1.0] - 2021.19.8

### LINKED ISSUE #22: Re/Connecting while in Chat
While using the website, i always seem to have the Problem of constant Connection loss, and with that, loosing the Chatpartner. Stable Internetconnection on my Side.

### LINKED ISSUE #26: select gender
no matter what gender I pick it wont let me find a partner, ive restarted browser and pc but nothing

### LINKED ISSUE #27: Connection issues
I have noticed that when i have a bump in my WiFi, like for a second i have no WiFi because bad connection, YiffSpot will go into this strange state where it thinks I have a partner, and doesn't think I have a partner at the same time.  
I get the message that I have a partner and i am in a session
I cannot type and post messages because I have no partner
I can't look for a new partner or block the non-existent partner I have
I sometimes get a message saying my non-existent partner is typing
this state after refreshing the page or closing my browser will last for several minutes (3-15)
have only tested for 5-10 minutes but the problem does not seem to go away unless I refresh or close my browser

### LINKED ISSUE #50: multiple active session problem
I keep loading up yiffspot and it says I have another active session but theres nothing else opened up. I shut down my browser and restarted my phone but nothing changed. the only abnormal thing I saw was over 300 people were online

### LINKED ISSUE #61: Getting errors when running the project
When i click the find partner button i get this error 

partner.js:121 Uncaught DOMException: Failed to execute 'send' on 'WebSocket': Still in CONNECTING state.
    at Object.findPartner [as find] (webpack:///./src/client/js/partner.js?:121:12)
    at HTMLFormElement.eval (webpack:///./src/client/js/index.js?:118:13)
findPartner @ partner.js:121
eval @ index.js:118

Can you please provide me exact setup instructions how to run the project locally using docker?

When i do npm install i get several errors too.

### LINKED ISSUE #85: Random Disconnect
Investigating.

### PR DESCRIPTION
**Pull request by xhayper**
### Changes
- Since reconnect, feature got added, If you open YiffSpot from another tab, The socket in the current tab will be closed.
### Fixes
- Fixed a bug where toast appears in the top right sometimes.
### Added
- Added reconnect feature, Give you 10 seconds to refresh your page if you got disconnected.
- Added Contributor Tag, So when I ask people to test stuff, they won't run away.
### Removed
- Removed `butterutil`, `utf-8-validate`, `slim-select` and `ws`.
### Replaced
- Replace `slim-select` with `tom-select`.
- Replace `ws` with `socket.io`.

Fixes #85
Close #61 ( Someone please tell them how to host it too)
Close #50 
Fixes #27
Close #26 
Fixes #22

### CODE REVIEW

File: src/client/js/index.js [hunk 1]
```diff
@@ -147,19 +162,43 @@ document.addEventListener('DOMContentLoaded', () => {
...
   });
 
-  // Handle when window is closed
-  window.addEventListener('beforeunload', () => socket.close());
+  window.onbeforeunload = function () {
+    user.setPartner(false);
+  }
 });
+
+// Gonna hide easter egg cuz yes :)
```
Comment (kisuka) [inline src/client/js/index.js:183]: remove please, it's unnecessary. sorry.

Comment (xhayper) [reply src/client/js/index.js:183]: Alright, Got it.

**Comment by xhayper** [comment]
Done

**Comment by xhayper** [comment]
I cleared out all of the unnecessary code now.

**Comment by xhayper** [comment]
All done.

**Comment by xhayper** [comment]
End.

**Pull request closed without merging**